ProcessLib/Process.h

```diff
@@ -258,24 +269,8 @@ class Process
 		}
 		assert(result && result->size() == _x->size());
 
-#ifdef USE_PETSC
-		std::unique_ptr<double[]> u(new double[_x->size()]);
-		_x->getGlobalVector(u.get());  // get the global solution
-
-		std::size_t const n = _mesh.getNNodes();
-		for (std::size_t i = 0; i < n; ++i)
-		{
-			MeshLib::Location const l(_mesh.getID(),
-			                          MeshLib::MeshItemType::Node, i);
-			auto const global_index = std::abs(  // 0 is the component id.
-			    _local_to_global_index_map->getGlobalIndex(l, 0));
-			(*result)[i] = u[global_index];
-		}
-#else
 		// Copy result
-		for (std::size_t i = 0; i < _x->size(); ++i)
-			(*result)[i] = (*_x)[i];
-#endif
+		_x->getValues(&(*result)[0]);
```

Comment:
In the nearest feature the `_x` vector will contain multiple components, which must be redistributed to different mesh properties. Will the proposed approach work?

Essentially it will be very similar to the previous version of the petsc code:
```c++
std::size_t const n = _mesh.getNNodes();
for (std::size_t i = 0; i < n; ++i)
{
	MeshLib::Location const l(_mesh.getID(),
	                          MeshLib::MeshItemType::Node, i);
	for (int component_id = 0; component_id < DisplacementDim;
	     ++component_id)
	{
		auto const global_index =
		    _local_to_global_index_map->getGlobalIndex(
		        l, component_id);
		(*result)[i * DisplacementDim + component_id] =
		    (*_x)[global_index]; /// <<< Copy _x to result in different order
	}
}
```